### PR TITLE
Reimplement/fix/complete implementation of scrollbars

### PR DESCRIPTION
**Pull request by rock3r**
This PR overhauls how scrollbars are implemented, fixing all known issues and implementing the features that were forgotten/missing.

* As before, scrollbars have:
  * Two visibility styles:
    * `AlwaysVisible`, where the scrollbars are always visible and have always the same thickness. This is the most similar to Win/Linux behavior on macOS but doesn't 100% match it.
    * `WhenScrolling`, scrollbars are only visible while scrolling
  * Two track click behaviors:
    * `NextPage`, clicking moves the viewport by a whole page, but not more
    * `JumpToSpot`, moves the viewport to the point that was clicked (may be more than a page)
* In standalone, the scrollbars use the defaults for each platform (macOS: `WhenScrolling` + `JumpToPoint`, Win/Linux: `AlwaysVisible` + `JumpToPoint`)
* In the bridge, we have lifted some platform logic in `ScrollbarHelper` to react to macOS system prefs changes, like Swing does. On Win/Linux it's always using the defaults, like in standalone.
* TabStrip scrollbars require a lot of special handling, we'll take care of them separately (#553)
* Scrollbars remain "active"/"visible" as long as the mouse moves over the content, like in Swing, if you set it up correctly
* `ScrollableContainer` API makes it all super easy for you, and handles overlaying or not the scroll bars over the scrollable content, like Swing does
* The `scrollbarContentSafePadding` function provides an optional padding to apply to scrollable content important parts to avoid them being overlapped by `WhenScrolling` scrollbars.

### TODO
 - [x] Fix WhenScrolling style on macOS standalone/bridge
   - Visibility does not animate in/out (today's regression)
   - Horizontal scrollbars not appearing?
 - [x] Fix/implement lingering when the mouse moves in all `ScrollableContainer` variants
 - [x] Fix stuff broken in the standalone sample (if any — Markdown editor?)
 - [x] Fix ScrollableContainer issues (layout bugs & crashes)
 - [x] File issues for minor/separate issues (tabStrip, text area, etc.)
 - [x] Investigate native crash in Swing Comparison tab in IDE sample — split out to #556
 - [x] Test on Windows/Linux standalone, fix issues there
 - [x] Test on Windows/Linux bridge, fix issues there
 - [x] Do the last round of holistic testing
 - [x] Decide whether we should revisit and clean up more in the future
   - Split out #558 #559 and #560

### CODE REVIEW

**Comment by rock3r** [comment]
@hamen @devkanro I am sorry this ended up being huge, I had to rewrite ~95% of everything scrollbars as the previous implementation was simply not working well, and/or not flexible enough.

Once this is merged we can ship 0.23.0, finally!

PS: there are some minor inconsistencies with Swing tracked in the linked issues in the description — they're all P2/P3 for what I am concerned, though.

**Comment by hamen** [comment]
No worries. We will take care of it on Monday 🙏🏻

**Review by hamen — APPROVED** [review]

File: ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/SwingBridgeService.kt [hunk 1]
```diff
@@ -19,14 +19,17 @@ import kotlin.time.Duration.Companion.milliseconds
...
             IntelliJApplication.lookAndFeelChangedFlow(scope),
-            MacScrollbarHelper.scrollbarVisibilityStyleFlow,
-            MacScrollbarHelper.trackClickBehaviorFlow,
+            scrollbarHelper.scrollbarVisibilityStyleFlow,
+            scrollbarHelper.trackClickBehaviorFlow,
         ) { _, _, _ ->
             tryGettingThemeData()
-        }.stateIn(scope, SharingStarted.Eagerly, BridgeThemeData.DEFAULT)
+        }
+            .stateIn(scope, SharingStarted.Eagerly, BridgeThemeData.DEFAULT)
```
[the remark this answers is not in the document]
Comment (rock3r) [reply ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/SwingBridgeService.kt:32]: Fixed by switching to ktfmt... which we can do soon :)

File: ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/theme/IntUiBridge.kt [hunk 2]
```diff
@@ -968,7 +968,7 @@ private fun readTextFieldStyle(): TextFieldStyle {
             TextFieldMetrics(
                 cornerSize = CornerSize(DarculaUIUtil.COMPONENT_ARC.dp / 2),
                 contentPadding = PaddingValues(horizontal = 9.dp, vertical = 2.dp),
-                minSize = DpSize(minimumSize.width, minimumSize.height),
+                minSize = DpSize(144.dp, minimumSize.height),
```
[the remark this answers is not in the document]
Comment (rock3r) [reply ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/theme/IntUiBridge.kt:971]: Yolo™

**Pull request merged**